### PR TITLE
Dashboard code cleanup

### LINKED ISSUE #22312: Cannot set the default dashboard background
@juliushaertl I cannot set the default background

```
Technical details
Remote Address: 172.19.0.1
Request ID: tTjAbsWgdBqacAPLl1Xe
Type: Exception
Code: 0
Message: Argument 2 passed to OCA\Dashboard\Controller\DashboardController::setBackground() must be of the type string, null given, called in /var/www/nextcloud/lib/private/AppFramework/Http/Dispatcher.php on line 169
File: /var/www/nextcloud/lib/private/AppFramework/Http/Dispatcher.php
Line: 110

Trace
#0 /var/www/nextcloud/lib/private/AppFramework/App.php(151): OC\AppFramework\Http\Dispatcher->dispatch(Object(OCA\Dashboard\Controller\DashboardController), 'setBackground')
#1 /var/www/nextcloud/lib/private/Route/Router.php(308): OC\AppFramework\App::main('OCA\\Dashboard\\C...', 'setBackground', Object(OC\AppFramework\DependencyInjection\DIContainer), Array)
#2 /var/www/nextcloud/lib/base.php(1009): OC\Route\Router->match('/apps/dashboard...')
#3 /var/www/nextcloud/index.php(37): OC::handleRequest()
#4 {main}
```

### PR DESCRIPTION
**Pull request by juliusknorr**
Addressing the comments by @skjnldsv from #22214

Fix #22312

### CODE REVIEW

**Review by MorrisJobke — APPROVED** [review]
Looks good 👍

File: apps/dashboard/src/App.vue [hunk 1]
```diff
@@ -58,7 +58,7 @@
 				<a v-if="isAdmin" :href="appStoreUrl" class="button">{{ t('dashboard', 'Get more widgets from the app store') }}</a>
 
 				<h3>{{ t('dashboard', 'Change background image') }}</h3>
-				<BackgroundSettings :background="background" @updateBackground="updateBackground" />
+				<BackgroundSettings :background="background" @update:background="updateBackground" />
```
Comment (skjnldsv) [inline apps/dashboard/src/App.vue:61]: Is the stored format that different?
Can't you just update the background with `:background.sync="background"` ?

File: apps/dashboard/lib/Controller/DashboardController.php [hunk 2]
```diff
@@ -127,8 +127,8 @@ public function updateLayout(string $layout): JSONResponse {
 	/**
 	 * @NoAdminRequired
 	 */
-	public function setBackground(string $type, string $value): JSONResponse {
-		$currentVersion = (int)$this->config->getUserValue($this->userId, 'dashboard', 'backgroundVersion', 0);
+	public function setBackground(string $type = 'default', string $value = ''): JSONResponse {
+		$currentVersion = (int)$this->config->getUserValue($this->userId, 'dashboard', 'backgroundVersion', '0');
```
Comment (eneiluj) [inline apps/dashboard/lib/Controller/DashboardController.php:131]: Nice, it prevents a crash when setting 'default images'.

**Review by julien-nc — APPROVED** [review]
All good.

**Review by jancborchardt — APPROVED** [review]

**Comment by eneiluj** [comment]
Rebased on master and removed background-color definitions.

**Pull request merged**